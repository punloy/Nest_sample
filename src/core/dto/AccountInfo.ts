export class AccountInfo {
    id: number;
    name: string
}
