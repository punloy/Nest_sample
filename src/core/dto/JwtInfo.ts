export class JwtInfo {
    id: number;
    data: {
        name: string,
    };
    jti?: string;
}
