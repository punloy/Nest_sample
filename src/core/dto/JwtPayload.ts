export class JwtPayload {
    id: number;
    data: {
        name: string,
    };
    jti?: string;
}
