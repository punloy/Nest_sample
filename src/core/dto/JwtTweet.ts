export class JwtTweet {
    id: number;
    data: {
        name: string,
        tittle: string,
        content: string
    };
    jti?: string;
}
