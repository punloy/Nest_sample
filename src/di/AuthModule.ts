import { Module } from '@nestjs/common';
import { PassportModule } from '@nestjs/passport';
import { TypeOrmModule } from '@nestjs/typeorm';
import { JwtStrategy } from '../core/jwt/JwtStrategy';
import { LocalUsertrategy } from '../core/jwt/LocalUserStrategy';
import { tweet } from '../core/index';
import { AuthService } from '../service/AuthService';

@Module({
    imports: [
        PassportModule,
        TypeOrmModule.forFeature(Object.values(tweet))
    ],
    providers: [
        AuthService,
        LocalUsertrategy,
        JwtStrategy,
    ]
})
export class AuthModule { }