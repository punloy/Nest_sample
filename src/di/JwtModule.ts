import { Module } from "@nestjs/common";
import { JwtService } from "../service/JwtService";

@Module({
    providers: [JwtService],
    exports: [JwtService]
})
export class JwtModule { }
