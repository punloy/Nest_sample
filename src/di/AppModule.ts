import { Module, ValidationPipe } from '@nestjs/common';
import { HealthModule } from './HealthModule';
import { APP_PIPE, APP_FILTER } from '@nestjs/core';
import config from "config";
import { TypeOrmModuleOptions } from "@nestjs/typeorm";
import { TweetDatabaseModule } from './TweetDatabaseModule';
import { HttpExceptionFilter } from '../core/filter/HttpExceptionFilter';
import { UserModule } from './UserModule';
import { AuthModule } from './AuthModule';
import { TweetModule } from './TweetModule';

const tweetOption: Partial<TypeOrmModuleOptions> = config.get("tweetDB");

@Module({
  imports: [
    TweetDatabaseModule.register(tweetOption),
    UserModule,
    TweetModule,
    AuthModule,
    HealthModule],
  providers: [
    {
      provide: APP_PIPE,
      useClass: ValidationPipe
    }, {
      provide: APP_FILTER,
      useClass: HttpExceptionFilter
    }
  ],
})
export class AppModule { }
