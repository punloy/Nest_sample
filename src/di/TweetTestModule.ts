import { DynamicModule, Module } from "@nestjs/common";
import { TypeOrmModuleOptions } from "@nestjs/typeorm";
import config from "config";
import { AppModule } from "./AppModule";
import { TweetDatabaseModule } from "./TweetDatabaseModule";

const tweetOptions: Partial<TypeOrmModuleOptions> = config.get("tweetDB");

@Module({})
export class TweetTestModule {
    public static register(): DynamicModule {
        return {
            module: TweetTestModule,
            imports: [
                TweetDatabaseModule.register(tweetOptions),
                AppModule
            ]
        };
    }
}
