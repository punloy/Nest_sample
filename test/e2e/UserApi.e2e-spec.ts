import { withTestApp, withJwt } from "../helper/TestApp";
import { TweetTestModule } from "../../src/di/TweetTestModule";
import { TestTweetData } from "../helper/index";
import { JwtPayload } from "../../src/core/dto/JwtPayload";

const imports = [TweetTestModule.register()];

describe("UsersApi (e2e)", () => {
    let jwtpayload1, jwtpayload2: JwtPayload;

    beforeAll(withTestApp(imports, async () => {
        jwtpayload1 = await TestTweetData.createTestUser({ userId: 1 });
        jwtpayload2 = await TestTweetData.createTestUser({ userId: 2 });
    }));

    it("/user/signin/ (POST) should 200", withTestApp(imports, async (app, request) => {
        const result = await request
            .post("/api/v1/user/signin")
            .send({
                name: "test",
                password: "pwd"
            })
            .expect(200);
        expect(result.body).toMatchSnapshot();
    }))

    it("/user/signin/ (POST) should 403", withTestApp(imports, async (app, request) => {
        const result = await request
            .post("/api/v1/user/signin")
            .send({
                name: "test user 1",
                password: "password"
            })
            .expect(403);
        expect(result.body).toMatchSnapshot();
    }))

    it("/user/login/ (POST) should 200", withTestApp(imports, async (app, request) => {
        const result = await request
            .post("/api/v1/user/login")
            .send({
                name: "test user 1",
                password: "password"
            })
            .expect(200);
        expect(result.body.data).toMatchSnapshot({ accessToken: expect.any(String) });
    }))

    it("/user/login/ (POST) should 403 1", withTestApp(imports, async (app, request) => {
        const result = await request
            .post("/api/v1/user/login")
            .send({
                name: "wrong user",
                password: "password"
            })
            .expect(403);
        expect(result.body).toMatchSnapshot();
    }))

    it("/user/login/ (POST) should 403 2", withTestApp(imports, async (app, request) => {
        const result = await request
            .post("/api/v1/user/login")
            .send({
                name: "test user 2",
                password: "wrongpwd"
            })
            .expect(403);
        expect(result.body).toMatchSnapshot();
    }))

})