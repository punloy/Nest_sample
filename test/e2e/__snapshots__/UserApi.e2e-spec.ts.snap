// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersApi (e2e) /user/login/ (POST) should 200 1`] = `
Object {
  "accessToken": Any<String>,
  "expiredIn": 28800,
  "tokenType": "Bearer",
}
`;

exports[`UsersApi (e2e) /user/login/ (POST) should 403 1 1`] = `
Object {
  "data": Object {},
  "message": "user not found",
}
`;

exports[`UsersApi (e2e) /user/login/ (POST) should 403 2 1`] = `
Object {
  "data": Object {},
  "message": "user password not correct",
}
`;

exports[`UsersApi (e2e) /user/signin/ (POST) should 200 1`] = `
Object {
  "data": Object {},
  "message": "success",
}
`;

exports[`UsersApi (e2e) /user/signin/ (POST) should 403 1`] = `
Object {
  "data": Object {},
  "message": "user account already existed",
}
`;
